libraries/architecture/src/main/kotlin/io/element/android/libraries/architecture/AsyncAction.kt

```diff
@@ -159,17 +160,25 @@ suspend inline fun <T> runUpdatingState(
         callsInPlace(resultBlock, InvocationKind.EXACTLY_ONCE)
     }
     state.value = AsyncAction.Loading
-    return resultBlock().fold(
-        onSuccess = {
-            state.value = AsyncAction.Success(it)
-            Result.success(it)
-        },
-        onFailure = {
-            val error = errorTransform(it)
-            state.value = AsyncAction.Failure(
-                error = error,
-            )
-            Result.failure(error)
-        }
-    )
+    try {
+        return resultBlock().fold(
+            onSuccess = {
+                state.value = AsyncAction.Success(it)
+                Result.success(it)
+            },
+            onFailure = {
+                val error = errorTransform(it)
+                state.value = AsyncAction.Failure(
+                    error = error,
+                )
+                Result.failure(error)
+            }
+        )
+    } catch (e: CancellationException) {
```

Comment:
I wonder if it wouldn't be a better solution to instead add a specific `catch (e: TimeoutCancellationException)` entry. My reasoning is:

1. We want to catch those exceptions most of the time, the only reason we don't is because they extend `CancellationException` which we want to rethrow so the coroutines work as expected.
2. We probably don't want to catch most `CancellationException`s, since these could happen i.e. when you exit a screen if some long task is running and could display error messages/dialogs while this screen transition is happening.

Reply:
Also, the `try` block should probably only wrap `resultBlock()` and not the code in the `fold` lambdas, even if in practice the code behavior should be quite similar.